packages/flutter_tools/lib/src/localizations/gen_l10n_templates.dart

```diff
@@ -112,12 +112,17 @@ abstract class @(class) {
 @(delegateClass)
 ''';
 
-const String numberFormatTemplate = '''
-    final intl.NumberFormat @(placeholder)NumberFormat = intl.NumberFormat.@(format)(
+const String numberFormatPositionalTemplate = '''
+    final @(placeholder)NumberFormat = intl.NumberFormat.@(format)(localeName);
+    final @(placeholder)String = @(placeholder)NumberFormat.format(@(placeholder));
```

Comment:
Follow Flutter style guide and specify the type of each variable. It seems that we unfortunately do not catch this in generated code. I'll file an issue.
```suggestion
    final intl.NumberFormat @(placeholder)NumberFormat = intl.NumberFormat.@(format)(localeName);
    final String @(placeholder)String = @(placeholder)NumberFormat.format(@(placeholder));
```

Reply:
Google's style recommendation has a section named `Avoid using var and dynamic`.
I think I got confused because the effective dart style says do not specify type where it can be inferred from RHS value.

I thought it was redundant to specify types because it can be inferred by RHS at the time of initialization.

Reply:
In the framework, we have the `always_specify_types` lint enabled: https://github.com/flutter/flutter/blob/master/analysis_options.yaml#L56

Reply:
Ok, will keep it in mind.

Reply:
Can't we just shorten it? I mean it can be reduced to a single line.

```
final intl.NumberFormat @(placeholder)NumberFormat = intl.NumberFormat.@(format)(localeName);
final String @(placeholder)String = @(placeholder)NumberFormat.format(@(placeholder));
```

can be reduced to

```
final String @(placeholder)String = intl.NumberFormat.@(format)(localeName).format(@(placeholder));
```

Reply:
I originally did it this way because `intl.NumberFormat.@(format)(localeName).format(@(placeholder));` is incredibly confusing to read. The generated code might be simpler to read, but it definitely looks weird given that there's interpolated strings, along with actual function arguments